Sage-Bionetworks/sage-monorepo
# feat(openchallenges): initialize basic docs site

Fixes #2395 

## Changelog
- add [MkDocs](https://www.mkdocs.org/) and [Material for MkDocs](https://squidfunk.github.io/mkdocs-material/) configurations
- "port" over some docs into the docs site
- add CI workflow for building docs

> [!WARNING]
> There's a lot going on in our current `docs` folder that may no longer be relevant and/or too complex to belong on the docs site.  I highly suggest we revisit the docs in another PR to re-organize and/or update the contents

## Preview

![docs](https://github.com/Sage-Bionetworks/sage-monorepo/assets/9377970/2f3bf03a-7ee0-400f-9b6e-325bbe430cc1)

## Local development

To test out this branch, you will need to:

1. install mkdocs, mkdocs-material, and termynal, ideally in a virtual environment (I used conda):

```
pip install mkdocs mkdocs-material termynal
```

2. run mkdocs serve:

```
mkdocs serve
```

This will open the docs at `http://127.0.0.1:8000/`

> [!NOTE]
> Because it uses the `8000` port, you should not be running the web app while testing out this feature.
